cppcache/src/PdxInstanceImpl.cpp

```diff
@@ -229,22 +225,13 @@ void PdxInstanceImpl::writeField(std::shared_ptr<PdxWriter> writer,
           dynamic_cast<CacheableStringArray*>(value.get());
       if (val != nullptr) {
         int size = val->length();
-        if (val->operator[](0)->isCString()) {
-          char** strings = new char*[size];
-          for (int item = 0; item < size; item++) {
-            strings[item] = const_cast<char*>(val->operator[](item)->asChar());
-          }
-          writer->writeStringArray(fieldName, strings, size);
-          delete[] strings;
-        } else if (val->operator[](0)->isWideString()) {
-          wchar_t** strings = new wchar_t*[size];
-          for (int item = 0; item < size; item++) {
-            strings[item] =
-                const_cast<wchar_t*>(val->operator[](item)->asWChar());
-          }
-          writer->writeWideStringArray(fieldName, strings, size);
-          delete[] strings;
+        char** strings = new char*[size];
```

Comment:
Probably but out of scope with this ticket. Was thinking of writing tickets to replace CacheableArray/WriteArray/ReadArray with vector since vector behaves more like Java arrays from which these come.